pyyoutube/client.py

```diff
@@ -127,6 +137,68 @@ def __init__(
                 )
             )
 
+    def _from_client_secrets_file(self, client_secret_path: str):
+        """Set credentials from client_sectet file
+
+        Args:
+            client_secret_path:
+                path to the client_secret.json file, provided by google console
+
+        Raises:
+            PyYouTubeException: missing required key, client_secret file not in 'web' format.
+        """
+        try:
+            secrets_data = None
+
+            with open(client_secret_path, "r") as f:
+                secrets_data = json.load(f)
+
+            # For now only 'web' client_secret files are support,
+            # some 'installed' type files can have missing 'client_secret' key
+            if "web" in secrets_data:
+                secrets_data = secrets_data["web"]
+
+                self.client_id = secrets_data["client_id"]
+                self.client_secret = secrets_data["client_secret"]
+
+                # Set default redirect to first defined in client_secrets file
+                if "redirect_uris" in secrets_data and len(secrets_data["redirect_uris"]) > 0:
+                    self.DEFAULT_REDIRECT_URI = secrets_data["redirect_uris"][0]
+
+                return
+
+            else:
+                raise PyYouTubeException(
+                    ErrorMessage(
+                        status_code=ErrorCode.INVALID_PARAMS,
+                        message="Only 'web' client_secret file are supported.",
+                    )
+                )
+
+        except KeyError as ke:
+            raise PyYouTubeException(
+                ErrorMessage(
+                    status_code=ErrorCode.INVALID_PARAMS,
+                    message=f"client_secret file is missing required key: '{ke.args[0]}'",
+                )
+            )
+
+        except Exception:
```

Comment:
I don't know if this should catch any  `Exception` and wrap it in a `PyYouTubeException`